### ISSUE TITLE
Handler Inteface

### ISSUE THREAD

**Issue by joshrtay** [weo-edu/store#1]
``` js
function *handleProfile(evt) {
  yield updateProfile(evt);
  this.emit('/user/changed'); // inter store comm
  this.send('/user/changed'); // communication with processes outside of data pipeline
}
```

**Comment by ashaffer** [comment]
No `this`.  We should use `broadcast`/`notify` functions.

**Comment by joshrtay** [comment]
can you give an example of broadcast and notify functions. i don't understand. also, broadcast and notify don't have a clear distinction to me.

i'm proposing `emit` and `send` because they draw on analogies from node. if we think of the entire data pipeline as a process then emits are how you to do intraprocess events and `send` is how you emit events to the outside world. https://nodejs.org/api/child_process.html#child_process_event_message

**Comment by joshrtay** [comment]
also, storm seems to be using `emit`

https://github.com/RallySoftware/node-storm

**Comment by ashaffer** [comment]
Ya those could be fine.  Main thing to me is that they not use `this`.  The main downside of `emit` to me is the confusion with a local EventEmitter, but maybe it not being on `this` will help with that.  I think I slightly prefer `broadcast` or `notify` to `send` for pushing things back to koa though.  `send` doesn't imply events to me, it seems more like just pushing data over a stream or something.

**Comment by joshrtay** [comment]
agreed on send. ya. i think `broadcast`, `notify` or `publish` would work.  where do these `emit` and `broadcast` functions come from and how are they wired to nsq?

**Comment by joshrtay** [comment]
also what about log? how do we capture logs?

**Comment by ashaffer** [comment]
I was thinking:

``` javascript
var broadcast = require('lib/broadcast');
var emit = require('lib/emit');
```

At the top of each store.  I think they should each be in charge of their own logging.  However, if we want to namespace the logs (by source), we could do something like how `debug` works:

``` javascript
var broadcast = require('lib/broadcast')('user')
var emit = require('lib/emit')('user')
```

EDIT:

Or possibly for slightly greater generality:

``` javascript
//...
broadcast.use(log('user'))
```

**Comment by joshrtay** [comment]
i like the use api. we should create the same thing for log.

**Comment by joshrtay** [comment]
still don't know how we deal with request ids. might not really be doable unless each request is it's own child_process, but that seems pretty crazy.

**Comment by ashaffer** [comment]
You mean deal with request id's for logging?  If we're really committing to generators, the best way might actually be to do the crazy generator thing:

``` javascript
const specialLogValue = uuid();

function *log() {
  const requestId = yield specialLogValue;
  // ...
}
```

``` javascript
function *dispatch() {
  var item = fn.next();
  if(item.value === specialLogValue)
    fn.next(requestId);
}
```
